externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/SparkOperation.scala

```diff
@@ -62,7 +62,7 @@ abstract class SparkOperation(opType: OperationType, session: Session)
   protected def cleanup(targetState: OperationState): Unit = state.synchronized {
     if (!isTerminalState(state)) {
       setState(targetState)
-      Option(getBackgroundHandle).foreach(_.cancel(true))
+      if (shouldRunAsync) Option(getBackgroundHandle).foreach(_.cancel(true))
```

Comment:
Wrong file? This is `SparkOperation`

Reply:
The `cleanup` of the `SparkOperation` doesn't check the `shouldRunAsync` to cancel the background handle. Therefore I fixed the logic in this pull request.

Reply:
Seems the check is dispensable, `Option(getBackgroundHandle)` is None when sync execute.

Reply:
@hddong, as you mentioned, the `cleanup` of the `FlinkOperation` doesn't also check the `shouldRunAsync`, right?

Reply:
@SteNicholas : I'm not clear for `FlinkOperation`, just mean the change here is unnecessary.